Foundation/JSONEncoder.swift

```diff
@@ -2081,21 +2101,32 @@ extension _JSONDecoder {
     fileprivate func unbox(_ value: Any, as type: Decimal.Type) throws -> Decimal? {
         guard !(value is NSNull) else { return nil }
 
+#if false
+        // Darwin implementation from overlay
+        // Attempt to bridge from NSDecimalNumber.
+        if let decimal = value as? Decimal {
+            return decimal
+        } else {
+            let doubleValue = try self.unbox(value, as: Double.self)!
+            return Decimal(doubleValue)
+        }
+#else
         // On Darwin we get (value as? Decimal) since JSONSerialization can produce NSDecimalNumber values.
         // FIXME: Attempt to grab a Decimal value if JSONSerialization on Linux produces one.
         let doubleValue = try self.unbox(value, as: Double.self)!
         return Decimal(doubleValue)
+#endif
     }
 
     fileprivate func unbox<T : Decodable>(_ value: Any, as type: T.Type) throws -> T? {
         let decoded: T
-        if T.self == Date.self {
+        if T.self == Date.self || T.self == NSDate.self {
```

Comment:
Same here — all of the `|| T.self == ...` lead to `as!`-casts which are not valid on Linux